From bfb95e4a60c27ec0f9bb4668cc6163f5cfb3e635 Mon Sep 17 00:00:00 2001
From: Georg Pfuetzenreuter <mail@georg-pfuetzenreuter.net>
Date: Fri, 6 Sep 2024 01:39:32 +0200
Subject: [PATCH] Return from #list command with 0

The "#list" command should not be treated as a failure when it is
both a legitimate request by the client, and correctly answered by the
server. It is commonly used for assessing whether a rsync endpoint is
healthy, having it return with a non-zero exit code causes misleading
error reports, and, in case of socket activation, failed service
instances on the server.

Signed-off-by: Georg Pfuetzenreuter <mail@georg-pfuetzenreuter.net>
---
 clientserver.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

Index: rsync-3.4.0/clientserver.c
===================================================================
--- rsync-3.4.0.orig/clientserver.c
+++ rsync-3.4.0/clientserver.c
@@ -1377,7 +1377,7 @@ int start_daemon(int f_in, int f_out)
 		rprintf(FLOG, "module-list request from %s (%s)\n",
 			host, addr);
 		send_listing(f_out);
-		return -1;
+		return 0;
 	}
 
 	if (*line == '#') {
